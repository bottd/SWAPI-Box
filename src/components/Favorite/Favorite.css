.favoritesBtn {
  background-color: transparent;
  border: 4px solid #fdde09;
  border-radius: 30px;
  color: #fdde09;
  cursor: pointer;
  font-family: starWars;
  font-size: 28px;
  margin: 0 20%;
  outline: 0;
  padding-bottom: 12px;
  text-align: center;
  text-decoration: none;
  transition: all 0.4s ease-in-out;
  width: 60%;
}

.favoritesBtn:hover {
  background-color: #fdde09;
  border: 4px solid black;
  color: black;
}

.favoritesToggle {
  background-color: #fdde09;
  border: 4px solid black;
  color: black;
}

.favoritesIcon {
  position: relative;
  top: 4px;
}

@media screen and (max-width: 1000px) {
  .favoritesBtn {
    margin: 20px 20% 10px 20%;
  }
}

@media screen and (max-width: 850px) {
  .favoritesBtn {
    margin: 0 10% 5px 10%;
    width: 80%;
  }
}

@media screen and (max-width: 580px) {
  .favoritesBtn {
    height: 110px;
  }
}
