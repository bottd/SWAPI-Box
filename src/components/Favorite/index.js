import React, { Component } from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";
import "./Favorite.css";

class Favorite extends Component {
  constructor() {
    super();
    this.state = {
      linkDisabled:
        window.location.pathname === "/" ||
        window.location.pathname.split("/").includes("favorites")
    };
  }

  handleClick = e => {
    if (this.state.linkDisabled) {
      e.preventDefault();
    }
  };

  render() {
    const { favoritesCount, selectedCategory } = this.props;
    return (
      <div className="Favorites">
        <NavLink
          to={`${selectedCategory}/favorites`}
          className="favoritesBtn"
          onClick={this.handleClick}
        >
          <span className="favoritesIcon">#</span> Favorite{" "}
          {selectedCategory !== "initial" ? selectedCategory : ""}:{" "}
          {favoritesCount}
        </NavLink>
      </div>
    );
  }
}

Favorite.propTypes = {
  toggleFavorites: PropTypes.func.isRequired,
  favoritesCount: PropTypes.number.isRequired,
  selectedCategory: PropTypes.string.isRequired
};

export default Favorite;
