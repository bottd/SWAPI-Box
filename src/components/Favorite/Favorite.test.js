import React from 'react';
import { shallow, mount } from 'enzyme';
import Favorite from './index';

describe('Favorite', () => {
  it('Renders like snapshot', () => {
    const wrapper = shallow(<Favorite/>);
    expect(wrapper).toMatchSnapshot()
  });
});

