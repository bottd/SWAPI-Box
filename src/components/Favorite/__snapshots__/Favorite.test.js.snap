// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Renders like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorite />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="favoritesBtn"
      >
        Favorites
      </button>,
      "className": "Favorites",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Favorites",
        "className": "favoritesBtn",
      },
      "ref": null,
      "rendered": "Favorites",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="favoritesBtn"
        >
          Favorites
        </button>,
        "className": "Favorites",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorites",
          "className": "favoritesBtn",
        },
        "ref": null,
        "rendered": "Favorites",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
